src/java.base/share/classes/java/time/Duration.java

```diff
@@ -582,6 +582,20 @@ public long get(TemporalUnit unit) {
     }
 
     //-----------------------------------------------------------------------
+    /**
+     * Checks if this duration is positive, excluding zero.
+     * <p>
+     * A {@code Duration} represents a directed distance between two points on
+     * the time-line and can therefore be positive, zero or negative.
+     * This method checks whether the length is greater than zero.
+     *
+     * @return true if this duration has a total length greater than zero
+     * @since 18
+     */
+    public boolean isPositive() {
+        return (seconds | nanos) > 0;
```

Comment:
I had to think whether this logic is correct, but I believe it is because `nanos` is 32 bits and positive so won't impact the negative bit of `seconds`.

Reply:
Thanks, Stephen. Yes, `nanos` is even smaller, less than `1,000,000,000`.